src/components/ViewRequest/Actions/ReworkRequest.tsx

```diff
@@ -64,12 +64,30 @@ const ReworkRequest = () => {
       myRoles.isOSF
     ) {
       disableReworkButton = false;
-    } else if (request.data?.subStage === "CAPackageReview" && myRoles.isCA) {
+    } else if (
+      request.data?.subStage === "SelectionPackageCAApproval" &&
+      myRoles.isCA
+    ) {
       disableReworkButton = false;
     } else if (currentStage?.key === "Recruiting" && myRoles.isCSF) {
       disableReworkButton = false;
     } else if (currentStage?.key === "Selection" && myRoles.isCSF) {
       disableReworkButton = false;
+    } else if (
+      currentStage?.key === "SelectionPackageCSFApproval" &&
+      myRoles.isCSF
+    ) {
+      disableReworkButton = false;
+    } else if (
+      currentStage?.key === "SelectionPackageHQApproval" &&
```

Comment:
All these SelectionPackage checks should be looking at request.data?.subStage rather than currentStage?.key, as they are subStages